### DISCUSSION TITLE
Question about BRDFS

### DISCUSSION THREAD

**Discussion by saeedhd96** [mitsuba-renderer/mitsuba3#430] [Q&A]
I have two questions regarding the BRDFs:

1. Is it possible to represent roughplastic `alpha` also using a texture or it is only float?
2. For the principled BSDF, what parameters should create perefectly diffuse? I can never get perfectly diffuse surface using principled.

For example:

```
import mitsuba as mi
scene = mi.load_dict(mi.cornell_box())
brdf_1 = mi.load_dict({'type': 'principled',
'base_color': {
    'type': 'rgb',
    'value': [0.8, 0.8, 0.8],
},
'flatness': 0.0,
'specular': 0.0,
'roughness': 1.0
})
brdf_2 = mi.load_dict({'type': 'diffuse',
'reflectance': {
    'type': 'rgb',
    'value': [0.8, 0.8, 0.8]
}})
rays = mi.Ray3f(o = 0.0, d = mi.warp.square_to_uniform_sphere(0.5))
si = scene.ray_intersect(rays)
s1 =  mi.Float(0.5)
s2 =  mi.Point2f(0.5)
_, principled_W = brdf_1.sample(mi.BSDFContext(), si, s1, s2)
_, diffuse_W = brdf_2.sample(mi.BSDFContext(), si, s1, s2)
print(principled_W)
print(diffuse_W)
```

The output of this is 
[[1.3195570707321167, 1.3195570707321167, 1.3195570707321167]]
[[0.800000011920929, 0.800000011920929, 0.800000011920929]]

which means the principled is not producing perfectly diffuse weights ...

**Comment by saeedhd96** [comment]
OK, I guess by definition it has two lobes and thus never will be perfectly diffuse.

**Comment by Speierers** [comment]
> Is it possible to represent roughplastic alpha also using a texture or it is only float?

This will depend on the BSDF. Take a look at the [plugin reference](https://mitsuba.readthedocs.io/en/stable/src/generated/plugins_bsdfs.html#the-principled-bsdf-principled). If the `Type` of the parameter says `texture`, it means that this parameter can be textured.

> For the principled BSDF, what parameters should create perefectly diffuse?

I believe you can make it perfectly diffuse by setting the `specular` parameter to `0.0`.

**Reply by saeedhd96** [reply]
But in my example above I have set the specular to 0.0 and it still produces non diffuse results. Could that be a bug then?

**Reply by Speierers** [reply]
According to this [diagram](https://mitsuba.readthedocs.io/en/stable/_images/principled.png), you will always get the retro-reflection with the diffuse. That's probably the reason why this doesn't match the simple `diffuse` BSDF.
